core/server/worker/src/main/java/alluxio/worker/grpc/GrpcExecutors.java

```diff
@@ -45,16 +45,9 @@
   public static final ExecutorService BLOCK_READER_EXECUTOR =
       new ImpersonateThreadPoolExecutor(new ThreadPoolExecutor(THREADS_MIN,
           ServerConfiguration.getInt(PropertyKey.WORKER_NETWORK_BLOCK_READER_THREADS_MAX),
-          THREAD_STOP_MS, TimeUnit.MILLISECONDS, new SynchronousQueue<>(),
+          THREAD_STOP_MS, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>(),
```

Comment:
Using a linked blocking queue will allow the worker to possibly become overloaded and requests could queue up for a while since the queue length is infinite.

Personally, I think it is better behavior from the user-experience side to get an immediate error back when the worker is overloaded with requests, rather than having to wait for a timeout.

That being said, I think it's ok to queue up requests, but I think there should be a bound on the queue where we start rejecting readers after a certain point in time. What are your thoughts?